sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client.py

```diff
@@ -73,21 +75,408 @@ def test_delete_repository(self, containerregistry_endpoint, containerregistry_r
+        repo = self.get_resource_name("repo")
+        tag = self.get_resource_name("tag")
+        self.import_image(HELLO_WORLD, ["{}:{}".format(repo, tag)])
+
+        client = self.create_registry_client(containerregistry_endpoint)
+
+        properties = client.get_manifest_properties(repo, tag)
+        received = client.update_manifest_properties(repo, tag, can_delete=False)
+        assert received.can_delete == False
+
+        received = client.update_manifest_properties(repo, tag, can_read=False)
+        assert received.can_read == False
+
+        received = client.update_manifest_properties(repo, tag, can_write=False)
+        assert received.can_write == False
+
+        received = client.update_manifest_properties(repo, tag, can_list=False)
+        assert received.can_list == False
+
+        received = client.update_manifest_properties(
+            repo, tag, can_delete=True, can_read=True, can_write=True, can_list=True
+        )
+
+        assert received.can_delete == True
+        assert received.can_read == True
+        assert received.can_write == True
+        assert received.can_list == True
+
+    @acr_preparer()
+    def test_get_tag(self, containerregistry_endpoint):
+        repo = self.get_resource_name("repo")
+        tag = self.get_resource_name("tag")
+        self.import_image(HELLO_WORLD, ["{}:{}".format(repo, tag)])
+
+        client = self.create_registry_client(containerregistry_endpoint)
+
+        properties = client.get_tag(repo, tag)
+
+        assert isinstance(properties, ArtifactTagProperties)
+        assert properties.name == tag
+
+    @acr_preparer()
+    def test_get_tag_does_not_exist(self, containerregistry_endpoint):
+        client = self.create_registry_client(containerregistry_endpoint)
+
         with pytest.raises(ResourceNotFoundError):
-            deleted_result = client.delete_repository("not_real_repo")
+            client.get_tag("Nonexistent", "Nonexistent")
 
     @acr_preparer()
-    def test_transport_closed_only_once(self, containerregistry_endpoint):
-        transport = RequestsTransport()
-        client = self.create_registry_client(containerregistry_endpoint, transport=transport)
-        with client:
-            for r in client.list_repository_names():
-                pass
-            assert transport.session is not None
+    def test_update_tag_properties(self, containerregistry_endpoint):
+        repo = self.get_resource_name("repo")
+        tag = self.get_resource_name("tag")
+        self.import_image(HELLO_WORLD, ["{}:{}".format(repo, tag)])
+
+        client = self.create_registry_client(containerregistry_endpoint)
+
+        properties = client.get_tag(repo, tag)
+        properties.can_delete = False
+        properties.can_read = False
+        properties.can_write = False
+        properties.can_list = False
+        received = client.update_tag_properties(repo, tag, properties)
```

Comment:
Can I combine this with kwargs? e.g.
```python
received = client.update_tag_properties(repo, tag, properties, can_delete=True)
```

Reply:
I have it so you can do one or the other. Ie.
```python
received = client.update_tag_properties(repo, tag, properties) # where properties has all the properties you want to update
# OR
received = client.update_tag_properties(repo, tag, can_delete=True)
```
In the situation where we supported both, properties has a `can_delete` property. Should we take the **kwarg as the actual one to use if they are different?